protected/humhub/modules/admin/views/setting/topics.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+use humhub\modules\admin\assets\AdminTopicAsset;
+use humhub\modules\admin\models\forms\GlobalTopicSettingForm;
+use humhub\modules\topic\models\Topic;
+use humhub\modules\ui\form\widgets\ActiveForm;
+use humhub\modules\ui\view\components\View;
+use humhub\widgets\Button;
+use humhub\widgets\GridView;
+use humhub\widgets\ModalButton;
+use yii\data\ActiveDataProvider;
+use yii\helpers\Html;
+use yii\widgets\Pjax;
+
+/**
+ * @var View $this
+ * @var ActiveDataProvider $dataProvider
+ * @var Topic $addModel
+ * @var GlobalTopicSettingForm $globalTopicSettingModel
+ * @var bool $suggestGlobalConversion
+ */
+
+AdminTopicAsset::register($this);
+
+?>
+<?php $this->beginContent('@admin/views/setting/_advancedLayout.php') ?>
+<?php Pjax::begin(['enablePushState' => false, 'id' => 'global-topics']); ?>
+<?php $form = ActiveForm::begin(['options' => ['data-pjax' => true]]); ?>
+<p><?= Yii::t('AdminModule.settings', 'Add topics that you will use in your posts. Topics can be personal interests or general terms. When posting, you can select them by choosing "Topics" and it will be easier for other users to find your posts related to that topic.') ?></p>
+
+<?= $form->field($addModel, 'name', [
+    'template' => '
+<div class="input-group">
+{input}
+<span class="input-group-btn">
+    ' . Button::defaultType()->icon('add')->loader()->submit() . '
+</span>
+</div>
+{error}
+{hint}',
+    'inputOptions' => [
+        'style' => 'height:36px',
+        'class' => 'form-control',
+        'placeholder' => Yii::t('AdminModule.settings', 'Add Topic'),
+    ],
+    'errorOptions' => ['style' => ['display' => 'inline-block'], 'tag' => 'span'],
+    'hintOptions' => ['style' => ['display' => 'inline-block'], 'tag' => 'span'],
+])->hint($suggestGlobalConversion
+    ? Button::info()->setText(Yii::t('AdminModule.settings', 'Convert \'{topicName}\' into global', ['topicName' => $addModel->name]))
+        ->action('admin.topic.convertTopic')
+        ->xs()->loader(false)
+        ->confirm(
+            Yii::t('AdminModule.settings', '<strong>Confirm</strong> topic conversion'),
+            Yii::t('AdminModule.settings', 'Do you really want to convert \'{topicName}\' into global?', ['topicName' => $addModel->name]),
+            Yii::t('base', 'Convert')
+        )
+    : '') ?>
+<?= Html::hiddenInput('convert-to-global', 0) ?>
+<?php ActiveForm::end(); ?>
+<?php $form = ActiveForm::begin([
+    'id' => 'global-topics-settings-form',
+    'options' => ['data-pjax' => true]
+]); ?>
+<?= $form->field($globalTopicSettingModel, 'restrictAdditionalTopics')
```

Comment:
I don't understand exactly when we need the checkbox and when we need the button. See screenshot.
![image](https://github.com/user-attachments/assets/452c7b4d-6f50-499a-be75-e5b6b075b22c)

As soon as there is a global topic, there should be no “local” topics with the same name. 

---
The ideal workflow for me would be:
- Initially, only the “Add” field is displayed. (green area) Without checkbox and button.


![image](https://github.com/user-attachments/assets/ef08007d-4e77-484c-ae04-0f2bd8610fa4)



- If the user enters a topic that has not yet been used, the topic is simply added to the list.

- If the topic already exists at the space level, it will appear after clicking on “+” and then the checkbox for confirmation. 

![image](https://github.com/user-attachments/assets/38dbfffb-c32e-47ad-b389-b212abb86d1b)

Reply:
Button is an action, it converts only currently typed/found this topic to global when there is conflict.

Checkbox is an option not an action, it will disable topic creation on space/user level, and also converts ALL topics to global. So after checkbox can be unchecked to allow topic creation on space/user level.

Reply:
@gevorgmansuryan Hmm, ok, I see. Can you remove the checkbox with the current functionality (disable  space/profile tags) for now. Maybe it will make sense in the future. So we might readd.